src/core/expression/predicate.visitor.js

```diff
@@ -70,25 +67,25 @@ export class PredicateVisitor extends Visitor {
 				predicate = l => isNull(l);
 				break;
 			case 'equals':
-				predicate = l => equals(rCastAs(l), l);
+				predicate = l => equals(parse(l), r);
```

Comment:
i can't see code where we parse r? don't we need to do it?

Reply:
Okay, you're absolutely right. I just was not able to realize why we should parse `r` to itself type. But `rt` technically is not equals to real `r` type and we really need to parse it everywhere. Thanks.